packages/vue-app/template/router.scrollBehavior.js

```diff
@@ -47,8 +45,12 @@ export default function (to, from, savedPosition) {
 
   const nuxt = window.<%= globals.nuxt %>
 
-  // triggerScroll is only fired when a new component is loaded
-  if (to.path === from.path && to.hash !== from.hash) {
+  if (
+    // Page hash changes
+    (to.path === from.path && to.hash !== from.hash) ||
+    // Initial load (vuejs/vue-router#3199)
+    to === from
```

Comment:
@posva Do you think it is a safe check for detecting initial load?

Reply:
not completely sure, we could also check if `from` is the `START` location that is exported in v4.

Reply:
I tried exposing [START](https://github.com/CoolCyberBrain/vue-router/blob/v3.2.0/src/util/route.js#L52) ref but checking here, on initial load `from` and `to` are same references but not equal to START:

![image](https://user-images.githubusercontent.com/5158436/90332714-992f8f80-dfbf-11ea-8f61-baa3aa51f0e0.png)